controllers/stardoginstance_controller.go

```diff
@@ -7,36 +7,221 @@ package controllers
 
 import (
 	"context"
+	"fmt"
+	"github.com/vshn/stardog-userrole-operator/stardogrest/stardogrestapi"
+	"net/url"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"time"
 
 	"github.com/go-logr/logr"
+	"k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
-	stardogv1alpha1 "github.com/vshn/stardog-userrole-operator/api/v1alpha1"
+	. "github.com/vshn/stardog-userrole-operator/api/v1alpha1"
+	"github.com/vshn/stardog-userrole-operator/stardogrest"
 )
 
 // StardogInstanceReconciler reconciles a StardogInstance object
 type StardogInstanceReconciler struct {
 	client.Client
-	Log    logr.Logger
-	Scheme *runtime.Scheme
+	Log               logr.Logger
+	Scheme            *runtime.Scheme
+	ReconcileInterval time.Duration
 }
 
+const instanceUserFinalizer = "finalizer.stardog.instance.users"
+const instanceRoleFinalizer = "finalizer.stardog.instance.roles"
+
 // +kubebuilder:rbac:groups=stardog.vshn.ch,resources=stardoginstances,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=stardog.vshn.ch,resources=stardoginstances/status,verbs=get;update;patch
 
 func (r *StardogInstanceReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("stardoginstance", req.NamespacedName)
+	namespace := req.NamespacedName
+	_ = r.Log.WithValues("stardoginstance", namespace)
```

Comment:
The reconciler instance exists only once. I don't think you should override the Log property for a single stardog instance reconciliation. If you want to provide the resource name in all log statements, you should create a copy and assign a Log property in the reconciliation Context. There's no guarantee that this reconcile method is called synchronously. It's rather executed from multiple worker threads with multiple reconciliations in parallel.